libpermian/plugins/anaconda_webui/hypervisor.py

```diff
@@ -0,0 +1,148 @@
+import subprocess
+import time
+import logging
+
+
+START_PORT = '42000'
```

Comment:
I'm not a big fan of this approach as 42000+ ports fall into range of "Registered Ports". Although ports in range 41798-42507 are currently unassigned (according to https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?&page=133) using them is not a clean solution as ports from range 49152-65535 should be used instead as they are "Dynamic Ports, also known as the Private or Ephemeral Ports" (according to https://www.rfc-editor.org/rfc/rfc6335.html). On the other hand, problem with these ports may be that if prerouting is configured using one of these ports and then some service attempts to start using the same port, the communication would be forwarded to the VM which is not desired. This could be avoided by opening such a port (claiming it) (e.g. by `sock = socket.socket(); sock.bind(('', 0))` which also uses random port, the port number is then obtained calling `sock.getsockname()[1]`) which would then prevent other (correctly behaving) SW using it. The problem here is that this cannot be simply done over SSH.

Also the iptables approach is not something that's desired to be used directly and can be unreliable due to other (network infrastructure) firewall rules which may block incoming traffic on the ports set up by this code. Also the hypervisor may use system management services such as firewalld (or even libvirt) which may interfere with such directly added rules. If firewall based solution is in the end decided to be the desired one, I'd vote for using firewalld instead of calling iptables directly as firewalld supports also ebtables and xtables.

Thinking about it more and more now, a much better solution could be to use SSH port forwarding as the SSH connection is needed (by libvirt) for connecting to the hypervisor anyway. It could work in a way that SSH connection could be made from the system where Permian is run to the hypervisor adding SSH port forwarding using `-L` option with argument in form of `$LOCAL_PORT:$VM_IP:$VM_PORT`. When connection is done to localhost:$LOCAL_PORT SSH opens a new TCP connection to $VM_IP:$VM_PORT and "tunnels" the traffic there. Single SSH process can be used to create multiple such tunnels and they can be added iteratively if SSH control file is used.

Reply:
It seems the START_PORT was forbidden and no longer has any use so it should be deleted.